metricbeat/beater/config.go

```diff
@@ -1,9 +1,29 @@
 package beater
 
-import "github.com/elastic/beats/libbeat/common"
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+)
 
 // Config is the root of the Metricbeat configuration hierarchy.
 type Config struct {
 	// Modules is a list of module specific configuration data.
-	Modules []*common.Config `config:"modules" validate:"required"`
+	Modules       []*common.Config    `config:"modules"`
+	ReloadModules ModulesReloadConfig `config:"reload.modules"`
```

Comment:
Hm, that didn't work as I expected. Because I have a default config, which I already load on the metricbeat level, the config is never nil. I plan to extract the reload part anyways. Ok if I do it in a second step?